src/Deanery.java

```diff
@@ -0,0 +1,111 @@
+import java.util.ArrayList;
+import java.util.Optional;
+import java.util.Scanner;
+import java.time.format.DateTimeFormatter;
+import java.time.LocalDateTime;
+
+public class Deanery {
+    static DateTimeFormatter dtf = DateTimeFormatter.ofPattern("dd.MM.yyyy HH:mm:ss");
+    static LocalDateTime now = LocalDateTime.now();
+
+    private static String createRaport(Student student, String roomNumber) {
+        return "------------------------------------------" +
+                "\n" +
+                String.format("%20s", "STUDENT ") + student.getID().toString() +
+                "\n" + "\n" +
+                "First Name" + String.format("%10s", " ") + "|" + String.format("%3s", " ") + student.getName() + "\n" +
+                "Second Name" + String.format("%9s", " ") + "|" + String.format("%3s", " ") + student.getSecondName() + "\n" + "\n" +
+                "Raport generated on " + dtf.format(now) + "\n" +
+                "Room number: " + roomNumber +
+                "\n" +
+                "------------------------------------------";
+    }
+
+    public static void serve(Student student, String roomNumber, int hour) {
+        if (!(roomNumber.equals(Rooms.A123.toString())||roomNumber.equals(Rooms.B123.toString())||roomNumber.equals(Rooms.C124.toString())||roomNumber.equals(Rooms.D124.toString()))) {
+            throw new RuntimeException("The room does not match any od Deanery's rooms");
+        }
+        if (Rooms.valueOf(roomNumber) == Rooms.A123) {
```

Comment:
The code looks a bit complex here -> as you can see there is a lot of code duplication. 
1. You don't need to make the Rooms.valueOf(roomNumber) == Rooms.XXX comparison for every student - you could just use the Rooms.valueOf(roomNumber) to get a proper enum value - e.g. Rooms.A123 and then put the logic of hours validation into the enum
2. Maybe instead of keeping the logic of which student can be served in which room in this class, a concrete Student class -> FullTime/PartTime could have this information? Then you would only need to check if the given room is in the acceptable rooms for that kind of student. Also - a String "typeOfStudy" would be useless then. Because Java runtime would know whether a student is a FullTime or a PartTime